### ISSUE TITLE
Ghost Busters II w/ MT-32 crashes Dosbox

### ISSUE THREAD

**Issue by Ekamekia** [dosbox-staging/dosbox-staging#3096]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.81.0-alpha-1907-g256822025

### What Operating System are you using?

macOS (Apple Silicon)

### If Other OS, please describe

MacOS Sonoma 14.0

### Relevant hardware info

Macbook Air M1

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Story Mode:
Vigo may be the scourge of Carpathia, but this game is the scourge of my existence. I had this game on my 286 back in the day. There was a bug in it that would cause it to lock up after playing digital sounds, either to the PC speaker, or the sound blaster, depending what it felt like doing that day. Low and behold Dosbox’s emulation is so good that the same thing happens. Fortunately Creative Labs created a patch for the game that fixes the issue, but now I’ve run into another problem. Once the game is patched it runs smoothy unless you are using MT-32 music on MacOS (0.81.0-alpha-1907-g256822025) (Arm) Dosbox crashes out when you get to the lab. It does not crash on Win x64 (0.81.0-alpha-1907-g25682) though.

TLDR: Ghost Busters II crashes on MacOS (0.81.0-alpha-1907-g256822025) but not Win x64 (0.81.0-alpha-1907-g25682) If you are using MT-32 music.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Obtain game from link
2. Extract all 5 ZIPs into folder
3. run GB2FIX.EXE to patch sound blaster issues.
4. (Configure Dosbox to use MT-32)
5. run command:  "GBII.COM EGA MT-32 MOUSE"
6. 

### Download URL of affected game or software

http://www.abandonia.com/en/downloadgame/326

### Your configuration

```ini
https://file.io/j76SKATbfkfd
```

### Provide a Log

```text
https://file.io/Ch4WHBTt4D5f
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by Grounded0** [comment]
Can you check against 0.80.1. That's what we officially support until 0.81.0 is released (soon).

**Comment by Ekamekia** [comment]
Same behavior in 0.80.1 "DosBox Staging quit unexpectedly" on Mac, Just fine in Windows.

**Comment by johnnovak** [comment]
Please paste the config and logs here, your external links no longer work ("The transfer you requested has been deleted.")

"Dosbox crashes out when you get to the lab" — how do we do that? Can you provide savegames and/or clear instructions how to achieve that? Please don't assume the person who picks the ticket up is familiar with the game.

Thanks.

**Comment by Ekamekia** [comment]
I've made a video on setting up the game files and patching the game then running it to the point that it crashes.

After its been setup and patched run "GBII.COM EGA MT32 MOUSE"
Press enter to skip each scene, eventually you will get to the first level of game play, you don't need to play, just wait to get hit 3 times and it will end, press enter to skip past the next two scenes, and the lab will show up for maybe a second before it crashes.

Sorry about the logs, moved them to my Google Drive

Dosbox Config: https://drive.google.com/file/d/11fQBlgv54yvHmnEgl6gXbU8g7Rr4ZJE1/view?usp=drive_link

MacOS Crash Log: https://drive.google.com/file/d/1LDV4RgZl5ZYcKC6MLK3cAvydUZvQ4j3i/view?usp=drive_link

Video Tutorial : https://youtu.be/np0I6LBhbX4

**Comment by kcgen** [comment]
Thanks for the reproduction steps, @Ekamekia :+1: 

Can you test the build produced from this CI run?

https://github.com/dosbox-staging/dosbox-staging/actions/runs/6780082523?pr=3100

When it finishes building, there should be a universal DMG available at the bottom of that page.

**Comment by johnnovak** [comment]
> Thanks for the reproduction steps, @Ekamekia.
> 
> Can you test the build produced from this CI run?
> 
> https://github.com/dosbox-staging/dosbox-staging/actions/runs/6780082523?pr=3100
> 
> When it finishes building, there should be a universal DMG available at the bottom of that page.

Yes, it will be supported.

**Comment by Ekamekia** [comment]
It works! Thanks 👍👍
